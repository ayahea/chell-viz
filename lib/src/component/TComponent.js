"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || Object.assign || function(t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
        s = arguments[i];
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
            t[p] = s[p];
    }
    return t;
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var CellContext_1 = require("../context/CellContext");
var PlotlyHelper_1 = require("../helper/PlotlyHelper");
var ReactHelper_1 = require("../helper/ReactHelper");
var defaultProps = {
    data: [[0], [0]],
    height: 400,
    padding: 0,
    pointColor: '#000000',
    width: 400,
};
exports.TComponent = ReactHelper_1.withDefaultProps(defaultProps, /** @class */ (function (_super) {
    __extends(TComponentClass, _super);
    function TComponentClass(props) {
        var _this = _super.call(this, props) || this;
        _this.onMouseSelect = function (cb) { return function (e) {
            if (e.points) {
                cb(e.points.map(function (point) { return point.pointNumber; }));
            }
        }; };
        _this.state = __assign({}, _this.state);
        return _this;
    }
    TComponentClass.prototype.render = function () {
        var _this = this;
        var _a = this.props, data = _a.data, height = _a.height, padding = _a.padding, pointColor = _a.pointColor, width = _a.width;
        var coords = new Float32Array(data.length * 2);
        data.forEach(function (ele, index) {
            coords[index * 2] = ele[0];
            coords[index * 2 + 1] = ele[1];
        });
        return (React.createElement("div", { id: "TComponent", style: { padding: padding } },
            React.createElement(CellContext_1.CellContext.Consumer, null, function (_a) {
                var addCells = _a.addCells;
                return (React.createElement(PlotlyHelper_1.PlotlyChart, { config: __assign({}, PlotlyHelper_1.defaultConfig, { modeBarButtons: [['zoomOut2d', 'zoomIn2d'], ['resetScale2d', 'autoScale2d'], ['select2d', 'pan2d']] }), data: [
                        {
                            marker: {
                                color: pointColor,
                            },
                            mode: 'markers',
                            type: 'scatter',
                            x: data.map(function (ele) { return ele[0]; }),
                            xaxis: 'x',
                            y: data.map(function (ele) { return ele[1]; }),
                            yaxis: 'y',
                        },
                    ], layout: __assign({}, PlotlyHelper_1.defaultLayout, { height: height,
                        width: width, yaxis: {
                            autorange: true,
                        } }), onSelectedCallback: _this.onMouseSelect(addCells) }));
            })));
    };
    return TComponentClass;
}(React.Component)));
