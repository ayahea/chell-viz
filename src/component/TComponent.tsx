import { T_SNE_DATA_TYPE } from 'chell';
import * as React from 'react';
import { CartesianGrid, Scatter, ScatterChart, XAxis, YAxis } from 'recharts';
import { withDefaultProps } from '../helper/ReactHelper';

const defaultProps = {
  data: [[0], [0]] as T_SNE_DATA_TYPE,
  height: 450,
  width: 450,
};

const initialState = {
  chartHeight: 0,
  chartWidth: 0,
};

type Props = {} & typeof defaultProps;
type State = typeof initialState;

export const TComponent = withDefaultProps(
  defaultProps,
  class TComponentClass extends React.Component<Props, State> {
    constructor(props: Props) {
      super(props);
      this.state = {
        ...this.state,
        chartHeight: Math.floor(0.9 * this.props.height),
        chartWidth: Math.floor(0.9 * this.props.width),
      };
    }

    public render() {
      const { data, height, width } = this.props;
      return (
        <div id="TComponent" style={{ height, padding: 15, width }}>
          <ScatterChart height={this.state.chartHeight} width={this.state.chartHeight}>
            <XAxis type="number" dataKey={'x'} />
            <YAxis type="number" dataKey={'y'} />
            <CartesianGrid />
            <Scatter
              data={
                data
                  ? data.map(ele => {
                      return { x: ele[0], y: ele[1] };
                    })
                  : []
              }
            />
          </ScatterChart>
        </div>
      );
    }
  },
);
