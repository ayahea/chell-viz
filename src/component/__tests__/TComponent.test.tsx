import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { TComponent } from '../TComponent';

test('Should match existing snapshot when given simple data.', () => {
  const data = [[1, 2], [3, 4], [5, 6]];
  expect(toJson(shallow(<TComponent data={data} />))).toMatchSnapshot();
});
