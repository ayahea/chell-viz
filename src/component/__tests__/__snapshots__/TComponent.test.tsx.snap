// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match existing snapshot when given simple data. 1`] = `
<div
  id="TComponentChart"
>
  <Chart
    allowEmptyRows={false}
    chartActions={null}
    chartEvents={Array []}
    chartPackages={
      Array [
        "corechart",
      ]
    }
    chartType="ScatterChart"
    chartVersion="current"
    columns={Array []}
    data={
      Array [
        Array [
          "x",
          "y",
        ],
        Array [
          1,
          2,
        ],
        Array [
          3,
          4,
        ],
        Array [
          5,
          6,
        ],
      ]
    }
    dateFormat={null}
    diffdata={null}
    graph_id="ScatterChart"
    height="400px"
    legend_toggle={true}
    loadCharts={false}
    loader={
      <div>
        Rendering Chart
      </div>
    }
    numberFormat={null}
    options={Object {}}
    rows={Array []}
    width="400px"
  />
</div>
`;
